### PR TITLE
Fix IstioD causing the Cortex cluster to no longer be reachable

### LINKED ISSUE #2332: IstioD is using too many resources when lots of pods are being added/churned
#### Description

Find a way to disable the monitoring of k8s resources so that istiod stops using so many resources.

#### Consequence

When istiod uses too much memory, it triggers an OOM event on the other pods collocated on the same instance such as the ingress gateways, the cluster autoscaler, the operator, etc. Or it can make k8s evict the other pods which is equally bad. This can render the cluster unusable.

### PR DESCRIPTION
**Pull request by RobertLucian**
Fixes #2332

---

checklist:

- [x] run `make test` and `make lint`
- [x] test manually (i.e. build/push all images, restart operator, and re-deploy APIs)

### CODE REVIEW

**Review by miguelvr — COMMENTED** [review]
AFAIK there are a more places referring to prometheus in the default namespace

From the top of my head, look for the autoscaler, activator, cluster health commands. But it is better to do a wide search for anything using the word "default"

File: manager/install.sh [hunk 1]
```diff
@@ -35,6 +35,10 @@ function main() {
 function cluster_up() {
   create_eks
 
+  echo -n "￮ creating namespaces "
+  setup_namespaces
```
Comment (deliahu) [inline manager/install.sh:39]: nit: move this right before `setup_configmap` and remove the `echo`s

File: manager/manifests/istio.yaml.j2 [hunk 2]
```diff
@@ -26,7 +30,23 @@ spec:
         resources:
           requests:
             cpu: 100m  # default is 500m
-            memory: 200Mi  # default is 2048Mi == 2Gi
+            memory: 700Mi  # default is 2048Mi == 2Gi
+        hpaSpec:
+          minReplicas: 1
+          maxReplicas: 5  # edit autoscaleEnabled in values if increasing this
```
Comment (deliahu) [inline manager/manifests/istio.yaml.j2:36]: nit: remove this comment (and the others like it)

**Review by deliahu — APPROVED** [review]

File: pkg/health/health.go [hunk 3]
```diff
@@ -30,6 +30,7 @@ import (
 	"github.com/cortexlabs/cortex/pkg/lib/parallel"
 	"github.com/cortexlabs/cortex/pkg/types/clusterconfig"
 	kapps "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (miguelvr) [inline pkg/health/health.go:33]: kcore

**Review by miguelvr — APPROVED** [review]

**Pull request merged**